firedrake/ensemble/ensemble.py

```diff
@@ -1,12 +1,32 @@
+from functools import wraps
 import weakref
 from itertools import zip_longest
 
 from firedrake.petsc import PETSc
+from firedrake.function import Function
+from firedrake.cofunction import Cofunction
 from pyop2.mpi import MPI, internal_comm
 
 __all__ = ("Ensemble", )
 
 
+def _ensemble_mpi_dispatch(func):
```

Comment:
`def _overload_comm_method`?

Also I think a docstring saying "Decorator that..." would be helpful. We want something that says what this decorator is.

Reply:
Done